### DISCUSSION TITLE
Use INET instead of the string for `last_login_ip` and such

### DISCUSSION THREAD

**Discussion by savchenko** [pallets-eco/flask-security#999] [Ideas]
With a help of a simple class decorator we can convert between Python's `ipaddress` and PostgreSQL `INET` ([reference](https://www.postgresql.org/docs/16/functions-net.html))

I wonder if it would be feasible to default `last_login_ip` and similar columns to that instead of the string.

**Comment by jwag956** [comment]
flask-security supports a few different ORMs as well as a few different DB backends - so that is a consideration for all column definitions. Also - in general I consider this mostly a 'logging' value - not used in computations or other computations.

**Reply by savchenko** [reply]
Got it. The reason I was considering it is that Postgres can do some pretty neat operations on IPs. Obviously, not something that would automagically apply to other DB drivers.

**Comment by jwag956** [comment]
Note that Flask-Security provides many 'hooks' via Flask signals - see: https://flask-security-too.readthedocs.io/en/stable/api.html#signals
so you could easily capture the login ip upon authentication and store it in your DB in whatever format for future use.
